core/src/test/java/org/elasticsearch/bootstrap/BootstrapCheckTests.java

```diff
@@ -137,6 +137,49 @@ public String errorMessage() {
         assertThat(suppressed[1], hasToString(containsString("second")));
     }
 
+    public void testHeapSizeCheck() {
+        final int initial = randomIntBetween(0, Integer.MAX_VALUE - 1);
+        final int max = randomIntBetween(initial + 1, Integer.MAX_VALUE);
+        final AtomicLong initialHeapSize = new AtomicLong(initial);
+        final AtomicLong maxHeapSize = new AtomicLong(max);
+
+        final BootstrapCheck.HeapSizeCheck check = new BootstrapCheck.HeapSizeCheck() {
+            @Override
+            long getInitialHeapSize() {
+                return initialHeapSize.get();
+            }
+
+            @Override
+            long getMaxHeapSize() {
+                return maxHeapSize.get();
+            }
+        };
+
+        try {
+            BootstrapCheck.check(true, Collections.singletonList(check), "testHeapSizeCheck");
+            fail("should have failed to initial heap size not equal to max heap size");
```

Comment:
I'll open another PR to convert all the try / catch asserts in this test suite to use `expectThrows`.